lib/rainforest/cli/runner.rb

```diff
@@ -153,5 +155,55 @@ def get_environment_id url
 
       return environment['id']
     end
+
+    def upload_app(app_source_url)
```

Comment:
I think it would be a good idea to add specs for each of the logical branches in this method to make sure nothing breaks in a refactor.